Data/IntMap/Internal.hs

```diff
@@ -1253,9 +1934,11 @@ maxViewWithKey t =
 
 minViewWithKey :: IntMap a -> Maybe ((Key, a), IntMap a)
 minViewWithKey t =
-  case t of Nil -> Nothing
-            Bin p m l r | m < 0 -> case go r of (result, r') -> Just (result, binCheckRight p m l r')
-            _ -> Just (go t)
+  case t of
+    Nil -> Nothing
+    Bin p m l r | m < 0 ->
+      case go r of (result, r') -> Just (result, binCheckRight p m l r')
+    _ -> Just (go t)
   where
```

Comment:
I suspect a custom triple type will be easier for GHC to unbox than a pair with a pair in it. Also, I seem to remember that you suggested I get lazier about producing the `Just` constructor for `Data.Map`. We should probably try to be consistent about that one way or the other.

```haskell
data MinViewWithKeyResult a = MVWKR {-# UNPACK #-} !Key a !(IntMap a)
minViewWithKey t =
  case t of
    Nil -> Nothing
    Bin p m l r | m < 0 ->
      Just $ case go r of MVWKR k a r' -> let !t' = binCheckRight p m l r' in ((k,a), t')
    _ -> Just $ case go t of MVWKR k a t' -> ((k,a), t')
```